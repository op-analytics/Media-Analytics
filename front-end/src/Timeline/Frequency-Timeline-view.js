import CircularProgress from '@material-ui/core/CircularProgress';
import {makeStyles} from '@material-ui/core/styles';
import Axios from 'axios';
import React, {useState} from 'react';
import Form from '../Shared/Form';
import LineCharts from '../Shared/LineChartsV2';

const API_URL =
  process.env.NODE_ENV === 'production' ? '/api' : process.env.REACT_APP_API_URL;

const useStyles = makeStyles(() => ({
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    overflow: 'hidden',
    width: '94%',
  },
}));

const formData = [
  { label: 'Words', name: 'words', required: true },
  { label: 'Year from', name: 'year_from', required: true },
  { label: 'Year to', name: 'year_to', required: true },
];

const cleanDataset = dataset => ({ title: dataset.word, data: dataset.data });

export default function Timeline() {
  const [timelineDatasets, setTimelineDatasets] = useState([]);
  const [loading, setLoading] = useState(false);

  const classes = useStyles();

  const onSubmitHandler = ({ year_from: yearFrom, year_to: yearTo, words }) => {
    const wordArray = words.split(',');
    setLoading(true);
    Axios.post(`${API_URL}/timeline/frequency`, {
      words: wordArray,
      year_from: yearFrom,
      year_to: yearTo,
    })
      .then(response => {
        const wordFrequencyData = response.data.data;
        setTimelineDatasets(wordFrequencyData);
        setLoading(false);
      })
      // eslint-disable-next-line no-console
      .catch(error => console.log(error.response));
  };

  return (
    <>
      <h3>Word Frequency Timeline</h3>
      <div className={classes.container}>
        <Form formData={formData} onSubmit={onSubmitHandler} />
        {loading ? (
          <CircularProgress />
        ) : (
          <LineCharts
            datasets={timelineDatasets.map(cleanDataset)}
            xAxisKey="year"
            yAxisKey="freq"
            tooltipItems={[
              { key: 'freq', title: 'freq' },
              { key: 'count', title: 'count' },
              { key: 'rank', title: 'rank' },
            ]}
          />
        )}
      </div>
    </>
  );
}
