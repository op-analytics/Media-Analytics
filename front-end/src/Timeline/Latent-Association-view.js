import CircularProgress from '@material-ui/core/CircularProgress';
import {makeStyles} from '@material-ui/core/styles';
import Axios from 'axios';
import React, {useState} from 'react';
import {CartesianGrid, Line, LineChart, ResponsiveContainer, Tooltip, XAxis, YAxis} from 'recharts';
import Form from '../Shared/Form';
import {createTooltip} from '../Shared/utils';

const useStyles = makeStyles(() => ({
  chartContainer: {
    width: '80vw',
    height: '50vh',
    flex: '0 1 auto',
    maxWidth: '1000px',
    paddingBottom: '6vh',
  },
  chartTitle: {
    textAlign: 'center',
    '&:first-letter': {
      textTransform: 'uppercase',
    },
  },
  tooltip: {
    width: '200px',
    margin: 0,
    lineHeight: '24px',
    border: '1px solid #f5f5f5',
    backgroundColor: 'hsla(0,0%,100%,.8)',
    padding: '10px',
  },
  tooltipLabel: {
    color: '#333',
  },
  tooltipLabelFirstWord: {
    color: '#777',
    '&:first-letter': {
      textTransform: 'uppercase',
    },
  },
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    overflow: 'hidden',
    width: '94%',
  },
}));

const API_URL =
  process.env.NODE_ENV === 'production' ? '/api' : process.env.REACT_APP_API_URL;

const formData = [
  { label: 'Concept 1', name: 'concept_1', required: true },
  { label: 'Concept 2', name: 'concept_2', required: true },
  { label: 'Year from', name: 'year_from', required: true },
  { label: 'Year to', name: 'year_to', required: true },
];

export default function Timeline() {
  const [data, setData] = useState();
  const [loading, setLoading] = useState(false);

  const classes = useStyles();

  const onSubmitHandler = ({
    year_from: yearFrom,
    year_to: yearTo,
    concept_1: concept1,
    concept_2: concept2,
  }) => {
    const concept1Array = concept1.split(',');
    const concept2Array = concept2.split(',');
    const yearFromInt = parseInt(yearFrom);
    const yearToInt = parseInt(yearTo);
    setLoading(true);
    Axios.post(`${API_URL}/timeline/latent-association`, {
      concept_1: concept1Array,
      concept_2: concept2Array,
      year_from: yearFromInt,
      year_to: yearToInt,
    })
      .then(response => {
        setData(response.data.data);
        setLoading(false);
      })
      // eslint-disable-next-line no-console
      .catch(error => console.log(error.response));
  };

  return (
    <>
      <h3>Latent association over time</h3>
      <div className={classes.container}>
        <Form formData={formData} onSubmit={onSubmitHandler} />
        {loading ? (
          <CircularProgress />
        ) : (
          data && (
            <div className={classes.chartContainer}>
              <ResponsiveContainer>
                <LineChart
                  data={data}
                  margin={{
                    top: 10,
                    right: 30,
                    left: 10,
                    bottom: 10,
                  }}
                >
                  <CartesianGrid strokeDasharray="3 3" />
                  <XAxis dataKey="year_range" tickMargin={15} />
                  <YAxis
                    dataKey="association"
                    tickFormatter={tickValue => tickValue.toFixed(2)}
                    domain={['dataMin', 'dataMax']}
                  />
                  <Tooltip
                    content={createTooltip(classes, [
                      {
                        key: 'association',
                        title: 'Association',
                        formatFunction: item => item.toFixed(5),
                      },
                    ])}
                  />
                  <Line
                    type="linear"
                    dataKey="association"
                    stroke="#8884d8"
                    fill="#8884d8"
                    strokeWidth={3}
                    dot={{ strokeWidth: 5 }}
                    activeDot={{
                      stroke: '#3F51B5',
                      strokeWidth: 7,
                      border: 'white',
                    }}
                  />
                </LineChart>
              </ResponsiveContainer>
            </div>
          )
        )}
      </div>
    </>
  );
}
