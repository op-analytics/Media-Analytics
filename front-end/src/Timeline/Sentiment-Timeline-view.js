import CircularProgress from '@material-ui/core/CircularProgress';
import { makeStyles } from '@material-ui/core/styles';
import Axios from 'axios';
import React, { useState } from 'react';
import LineCharts from '../Shared/LineChartsV2';
import Form from './components/Timeline-form';

const API_URL =
  process.env.NODE_ENV === 'production'
    ? '/api'
    : process.env.REACT_APP_API_URL;

const useStyles = makeStyles(() => ({
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    overflow: 'hidden',
    width: '94%',
  },
}));

const cleanDataset = dataset => ({ title: dataset.word, data: dataset.data });

export default function Timeline() {
  const [timelineDatasets, setTimelineDatasets] = useState([]);
  const [loading, setLoading] = useState(false);

  const classes = useStyles();

  const onSubmitHandler = (e, yearFrom, yearTo, words) => {
    e.preventDefault();
    const wordArray = words.split(',');
    setLoading(true);
    Axios.post(`${API_URL}/timeline/sentiment`, {
      words: wordArray,
      year_from: yearFrom,
      year_to: yearTo,
    })
      .then(response => {
        const wordFrequencyData = response.data.data;
        setTimelineDatasets(wordFrequencyData);
        setLoading(false);
      })
      // eslint-disable-next-line no-console
      .catch(error => console.log(error.response));
  };

  return (
    <>
      <h3>Sentiment Over Timeline</h3>
      <div className={classes.container}>
        <Form onSubmitHandler={onSubmitHandler} />
        {loading ? (
          <CircularProgress />
        ) : (
          <LineCharts
            datasets={timelineDatasets.map(cleanDataset)}
            xAxisKey="year"
            yAxisKey="sentiment"
            tooltipItems={[{ key: 'sentiment', title: 'Sentiment' }]}
          />
        )}
      </div>
    </>
  );
}
