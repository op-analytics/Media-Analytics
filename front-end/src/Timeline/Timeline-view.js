import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Axios from 'axios';
import Form from './components/Timeline-form';
import LineCharts from '../Shared/LineCharts';

const API_URL = process.env.NODE_ENV === 'production' ? '/api/' : process.env.REACT_APP_API_URL;

const useStyles = makeStyles(() => ({
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    overflow: 'hidden',
    width: '94%',
  },
}));

export default function Timeline() {
  const [timelineDatasets, setTimelineDatasets] = useState([]);

  const classes = useStyles();

  const getLabels = wordFrequencyData =>
    wordFrequencyData[0].data.reduce((yearLabels, { year }) => yearLabels.concat(year * 1), []);

  const createDataset = ({ data, word }) => {
    const dataWithY = data.reduce(
      (accum, {
        wordFreq, rank, wordCount, year,
      }) =>
        accum.concat({
          word,
          year,
          wordFreq,
          rank,
          wordCount,
          y: wordFreq,
        }),
      [],
    );

    return {
      label: word,
      fill: false,
      lineTension: 0.1,
      backgroundColor: 'rgba(75,192,192,0.4)',
      borderColor: 'rgba(75,192,192,1)',
      pointBorderColor: 'rgba(172,75,125,0.8)',
      pointBackgroundColor: 'rgba(172,75,125,1)',
      pointBorderWidth: 1.5,
      pointHoverRadius: 5,
      pointHoverBackgroundColor: 'rgba(75,192,192,1)',
      pointHoverBorderWidth: 2,
      pointHitRadius: 10,
      data: dataWithY,
    };
  };

  const onSubmitHandler = (e, yearFrom, yearTo, words) => {
    e.preventDefault();
    const wordArray = words.split(',');
    Axios.post(`${API_URL}timeline/frequency`, {
      words: wordArray,
      year_from: yearFrom,
      year_to: yearTo,
    })
      .then(response => {
        const wordFrequencyData = response.data.data;
        setTimelineDatasets(
          wordFrequencyData.reduce(
            (accum, data) =>
              accum.concat({
                labels: getLabels(wordFrequencyData),
                datasets: [createDataset(data)],
              }),
            [],
          ),
        );
      })
      // eslint-disable-next-line no-console
      .catch(error => console.log(error.response));
  };

  return (
    <>
      <h3>Word Frequency Timeline</h3>
      <div className={classes.container}>
        <Form onSubmitHandler={onSubmitHandler} />
        <LineCharts chartDatasets={timelineDatasets} />
      </div>
    </>
  );
}
