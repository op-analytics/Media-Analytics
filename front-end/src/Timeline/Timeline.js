import React, { useState, useEffect } from "react";
import { Line } from "react-chartjs-2";
import Form from "./components/Form";
import { makeStyles, createMuiTheme } from "@material-ui/core/styles";
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles(theme => ({
  LineChart: {
    maxWidth: "1000px"
  }
}));

export default function Timeline() {
  const [data, setData] = useState({
    labels: [],
    datasets: []
  });

  const [datasets] = useState([
    {
      label: "My First dataset",
      fill: false,
      lineTension: 0.1,
      backgroundColor: "rgba(75,192,192,0.4)",
      borderColor: "rgba(75,192,192,1)",
      pointBorderColor: "rgba(172,75,125,0.8)",
      pointBackgroundColor: "rgba(172,75,125,1)",
      pointBorderWidth: 1.5,
      pointHoverRadius: 5,
      pointHoverBackgroundColor: "rgba(75,192,192,1)",
      pointHoverBorderWidth: 2,
      pointHitRadius: 10,
      data: [
        {
          year: 1980,
          word: "boat",
          rank: 1525,
          count: 3223,
          y: 0.0126
        },
        { year: 1981, word: "boat", rank: 1400, count: 3400, y: 0.013 },
        { year: 1982, word: "boat", rank: 1420, count: 3375, y: 0.01352324 },
        { year: 1983, word: "boat", rank: 1475, count: 4000, y: 0.01145463 },
        { year: 1984, word: "boat", rank: 1600, count: 3955, y: 0.012745 }
      ]
    }
  ]);

  const [labels] = useState(["1980", "1981", "1982", "1983", "1984"]);

  const classes = useStyles();

  useEffect(() => {
    setData(data => {
      return { ...data, datasets, labels };
    });
  }, [datasets, labels]);

  const createLabels = labels => {
    return function(tooltipItem, data) {
      return labels.reduce(
        (accum, { key, value }) =>
          accum.concat(
            key + data["datasets"][0]["data"][tooltipItem["index"]][value]
          ),
        []
      );
    };
  };

  return (
    <>
      <h3>Word Frequency Timeline</h3>
      <Grid
        container
        spacing='2'
      >
        <Grid item xs={'false'} lg={4}>
        </Grid>
        <Grid item xs={6} lg={4} spacing-xs-2>
          <Form />
        </Grid>
        <Grid item xs={'false'} lg={4}>
        </Grid>
        <Grid item xs={'false'} lg={3}>
        </Grid>
        <Grid item xs={12} lg={6} className="classes.LineChart" wrap={'wrap'}>
          <Line
            data={data}
            options={{
              maintainAspectRatio: true,
              responsive: true,
              tooltips: {
                callbacks: {
                  title: function(tooltipItem, data) {
                    const response =
                      data.datasets[0].data[tooltipItem[0].index];
                    return response["year"] + " - " + response["word"];
                  },
                  label: createLabels([
                    { key: "Frequency: ", value: "y" },
                    { key: "Count: ", value: "count" },
                    { key: "Rank: ", value: "rank" }
                  ])
                }
              }
            }}
          />
        </Grid>
      </Grid>
    </>
  );
}
