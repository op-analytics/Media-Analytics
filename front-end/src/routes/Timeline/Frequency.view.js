import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CircularProgress from '@material-ui/core/CircularProgress';
import FormControl from '@material-ui/core/FormControl';
import Grid from '@material-ui/core/Grid';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Autocomplete from '@material-ui/lab/Autocomplete';
import ToggleButton from '@material-ui/lab/ToggleButton';
import ChipInput from 'material-ui-chip-input';
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import LineCharts from '../../components/LineCharts';
import { getFrequencies } from '../../state/ducks/timeline';

const useStyles = makeStyles(theme => ({
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    overflow: 'hidden',
    width: '94%',
    paddingTop: '2rem',
    paddingBottom: '1rem',
  },
  label: {
    backgroundColor: 'white',
  },
  form: {
    width: '100%',
    height: '100%',
    padding: '2rem',
  },
  input: {
    border: 'solid 1px black',
    width: '100%',
  },
  formControl: {
    margin: theme.spacing(1),
    width: '100%',
  },
  submitButton: {
    width: '100%',
  },
  toggleButton: {
    maxWidth: '150px',
  },
  Card: {
    width: '45vw',
    minWidth: '400px',
  },
}));

const mediaOutlets = [
  { title: 'New York Times', name: 'New York Times', value: 'nyt' },
  { title: 'Wall Street Journal', name: 'Wall Street Journal', value: 'wsj' },
  { title: 'The Guardian', name: 'The Guardian', value: 'guardian' },
  { title: 'HuffPost', name: 'HuffPost', value: 'hp' },
];

const displayOptions = [
  { name: 'On a single chart', value: 'single' },
  { name: 'On individual charts', value: 'multiple' },
  { name: 'Grouped by media outlet', value: 'byOutlet' },
  { name: 'Grouped by word', value: 'byWord' },
];

/**
 * The frequency timeline page component
 * @component
 */
function Timeline() {
  const classes = useStyles();
  const dispatch = useDispatch();
  const [yearFrom, setYearFrom] = useState('');
  const [yearTo, setYearTo] = useState('');
  const [yAxisKey, setYAxisKey] = useState('freq');
  const [absolute, setAbsolute] = useState(false);
  const [outlets, setOutlets] = useState([]);
  const [displayOption, setDisplayOption] = useState('multiple');
  const [words, setWords] = useState([]);
  const [formSubmitted, setFormSubmitted] = useState(true);
  const loading = useSelector(state => state.timeline.loading);
  const frequencies = useSelector(state => state.timeline.frequencies);

  const minYear = 1970;
  const maxYear = 2020;

  const onSubmitHandler = e => {
    e.preventDefault();
    setFormSubmitted(true);
    dispatch(getFrequencies(words, minYear, maxYear, outlets));
  };

  return (
    <>
      <h3>Word Frequency Timeline</h3>
      <div className={classes.container}>
        <Card className={classes.Card}>
          <form className={classes.form} onSubmit={onSubmitHandler}>
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <FormControl className={classes.formControl}>
                  <ChipInput
                    label="Words:"
                    name="words"
                    newChipKeyCodes={[13, 32]} // Make new chip on enter and space key codes
                    blurBehavior="add" // Fix android chrome bug
                    onChange={newWords => {
                      const newWordExists = !newWords.every(item =>
                        words.includes(item),
                      );
                      if (newWordExists && outlets.length) {
                        setFormSubmitted(false);
                      }
                      setWords(newWords);
                    }}
                    required={!words.length}
                  />
                </FormControl>
              </Grid>
              <Grid item xs={12}>
                <FormControl className={classes.formControl}>
                  <Autocomplete
                    multiple
                    id="tags-standard"
                    options={mediaOutlets}
                    getOptionLabel={option => option.title}
                    filterSelectedOptions
                    onChange={(_, value) => {
                      const newOutlets = value.map(({ value: code }) => code);
                      const newOutletExists = !newOutlets.every(item =>
                        outlets.includes(item),
                      );
                      if (newOutletExists && words.length) {
                        setFormSubmitted(false);
                      }
                      setOutlets(newOutlets);
                    }}
                    renderInput={params => (
                      <TextField
                        {...params}
                        variant="standard"
                        label="Media Outlets"
                        placeholder="Add a media outlet"
                        required={!outlets.length}
                      />
                    )}
                  />
                </FormControl>
              </Grid>
              <Grid item md={3} sm={6} xs={12}>
                <FormControl className={classes.formControl}>
                  <TextField
                    className={classes.textField}
                    type="number"
                    label="Year from:"
                    name="year_from"
                    InputProps={{ inputProps: { min: minYear, max: maxYear } }}
                    onChange={e => {
                      const newYear = Number(e.target.value);
                      if (newYear >= minYear && newYear <= maxYear) {
                        setYearFrom(newYear);
                      }
                    }}
                    required
                  />
                </FormControl>
              </Grid>
              <Grid item md={3} sm={6} xs={12}>
                <FormControl className={classes.formControl}>
                  <TextField
                    className={classes.textField}
                    type="number"
                    label="Year to:"
                    name="year_to"
                    InputProps={{ inputProps: { min: minYear, max: maxYear } }}
                    onChange={e => {
                      const newYear = Number(e.target.value);
                      if (newYear >= minYear && newYear <= maxYear) {
                        setYearTo(newYear);
                      }
                    }}
                    required
                  />
                </FormControl>
              </Grid>
              {
                // Custom options start here
              }
              <Grid item md={3} sm={6} xs={12}>
                <FormControl className={classes.formControl}>
                  <InputLabel
                    className={classes.label}
                    id="y-axis-key-select-label"
                  >
                    Y-Axis Key
                  </InputLabel>
                  <Select
                    labelId="y-axis-key-select-label"
                    id="y-axis-key-select"
                    value={yAxisKey}
                    onChange={e => setYAxisKey(e.target.value)}
                  >
                    <MenuItem value="freq">Frequency</MenuItem>
                    <MenuItem value="count">Count</MenuItem>
                    <MenuItem value="rank">Rank</MenuItem>
                  </Select>
                </FormControl>
              </Grid>
              <Grid item md={3} sm={6} xs={12}>
                <FormControl className={classes.formControl}>
                  <ToggleButton
                    selected={absolute}
                    value={absolute}
                    onChange={() => {
                      setAbsolute(!absolute);
                    }}
                    className={classes.ToggleButton}
                  >
                    Display Absolute
                  </ToggleButton>
                </FormControl>
              </Grid>
              <Grid item md xs={12}>
                <FormControl className={classes.formControl}>
                  <InputLabel
                    className={classes.label}
                    id="display-option-select-label"
                  >
                    Display Options
                  </InputLabel>
                  <Select
                    labelId="display-option-select-label"
                    id="display-option-select"
                    value={displayOption}
                    onChange={e => setDisplayOption(e.target.value)}
                  >
                    {displayOptions.map(option => (
                      <MenuItem key={option.value} value={option.value}>
                        {option.name}
                      </MenuItem>
                    ))}
                  </Select>
                </FormControl>
              </Grid>
              {
                // Submit button
              }
              <Grid container item sm={12} justify="center">
                <Grid item xs={3} align="center">
                  <Button
                    variant="contained"
                    color="primary"
                    type="submit"
                    className={classes.submitButton}
                    margin="0 auto"
                    disabled={formSubmitted}
                  >
                    Submit
                  </Button>
                </Grid>
              </Grid>
            </Grid>
          </form>
        </Card>

        {loading ? (
          <CircularProgress />
        ) : (
          formSubmitted && (
            <LineCharts
              datasets={frequencies}
              xAxisKey="year"
              yAxisKey={yAxisKey}
              displayAbsolute={absolute}
              tooltipItems={[{ key: 'rank', title: 'rank' }]}
              words={words}
              mediaOutlets={outlets}
              allMediaOutlets={mediaOutlets}
              yearFrom={yearFrom}
              yearTo={yearTo}
              displayOption={displayOption}
            />
          )
        )}
      </div>
    </>
  );
}

export default Timeline;
