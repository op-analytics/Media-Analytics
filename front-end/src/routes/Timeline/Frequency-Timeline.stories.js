import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Timeline from './Frequency.view';

const mockStore = configureStore();

const initialState = {
  timeline: {
    frequencies: [
      {
        title: 'cat',
        data: [
          { year: 1990, rank: 400, count: 1000, freq: 0.083 },
          { year: 1991, rank: 400, count: 800, freq: 0.063 },
        ],
      },
    ],
  },
};

const store = mockStore(initialState);

export default {
  title: 'Timeline/Frequency',
  component: Timeline,
  // eslint-disable-next-line react/display-name
  decorators: [storyFn => <Provider store={store}>{storyFn()}</Provider>],
};

export const FrequencyTimeline = () => <Timeline />;
