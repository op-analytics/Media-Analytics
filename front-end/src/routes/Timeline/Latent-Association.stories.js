import { action,createStore, StoreProvider } from 'easy-peasy';
import React from 'react';

import Timeline from './Latent-Association.view';

const initialState = {
  timeline: {
    getAssociations: action(() => ({})),
    associations: [{ year_range: '1990-1994', association: 0.3209838560784504 }],
  },
};
const store = createStore(initialState);

export default {
  title: 'Timeline/LatentAssociation',
  component: Timeline,
  // eslint-disable-next-line react/display-name
  decorators: [
    storyFn => <StoreProvider store={store}>{storyFn()}</StoreProvider>,
  ],
};

export const LantentAssociationTimeline = () => <Timeline />;
