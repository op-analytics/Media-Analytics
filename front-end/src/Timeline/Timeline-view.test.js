import React from 'react';
import { shallow } from 'enzyme';
import Timeline from './Timeline-view';

describe('Timeline', () => {
  it('should render a form', () => {
    const component = shallow(<Timeline />);
    expect(component.find('Form').exists());
  });
});