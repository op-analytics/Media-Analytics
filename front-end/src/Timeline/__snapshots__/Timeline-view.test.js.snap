// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should render correctly 1`] = `
<Fragment>
  <h3>
    Word Frequency Timeline
  </h3>
  <div
    className="makeStyles-container-1"
  >
    <Form
      onSubmitHandler={[Function]}
    />
    <LineCharts
      chartDatasets={Array []}
    />
  </div>
</Fragment>
`;
